packages/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -12,17 +13,42 @@ import type { DialogProps, DialogState } from './Dialog.types';
  * @param ref - reference to root HTMLElement of Dialog
  */
 export const useDialog_unstable = (props: DialogProps, ref: React.Ref<HTMLElement>): DialogState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { overlay, open = false, type = 'modal' } = props;
+  const isNonModal = type === 'non-modal';
+
+  const contentRef = React.useRef(null);
+  const { modalAttributes } = useModalAttributes({ trapFocus: !isNonModal && true });
```

Comment:
There's also another return `triggerAttributes`. As I mentioned in the spec PR, it's necessary to revert focus when the dialog is dismissed.  `triggerAttributes` can help you do this. 

Feel free to propose another way of doing this. I guess you could also store `document.activeElement` in your first focusable effect. But we should make sure that this behaviour works from the beginning

Reply:
A trigger pattern might look something like this: https://www.radix-ui.com/docs/primitives/components/alert-dialog